app/Http/Controllers/TransactionController.php

```diff
@@ -81,9 +82,29 @@ public function show($id)
      * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function edit($id)
+
+    private function getLastTransaction($account_id, $transaction_id){
+        $transaction = Transaction::where('customer_id', $account_id)->get();
+        $lastTransaction = [];
+        for($x = 0 ; $x <  count($transaction); $x++){
+            if($transaction[$x]->id == $transaction_id){
+                try{
+                    $lastBalance = $transaction[$x - 1];
+                }catch(ErrorException $e){
+                    $lastBalance = $transaction[$x];
+                }
+            }
+        }
+        return $lastBalance;
+    }
+    private function getTransactionToBeUpdate($transaction_id){
+        return Transaction::find($transaction_id);
+    }
```

Comment:
I think this function is not necessary as you can directly call this code on edit function.